specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-05-05-preview/AlertsManagement.json

```diff
@@ -2293,6 +2337,34 @@
           "description": "tags to be updated"
         }
       }
+    },
+    "alertsMetaData": {
+      "description": "alert meta data information.",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/alertsMetaDataProperties"
+        }
+      }
+    },
+    "alertsMetaDataProperties": {
+      "description": "alert meta data property bag",
+      "properties": {
+        "metadataIdentifier": {
+          "type": "string",
+          "description": "Identification of the information to be retrieved by API call"
+        },
+        "data": {
+          "description": "Array of operations",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/alertsMetaDataValue"
+          }
+        }
+      }
+    },
+    "alertsMetaDataValue": {
```

Comment:
can we have strongly typed schema for this? from example it looks like when type is monitor service, the properties are name and displayName. You can have different schema when you introduce new type. it will make it more usable.

Reply:
@ravbhatnagar how to achieve this strongly typed schema? by using discriminator?

Reply:
Done